### PR TITLE
Update data.py

### PR DESCRIPTION
**Pull request by shayan137**

### CODE REVIEW

File: data.py [hunk 1]
```diff
@@ -348,24 +350,22 @@ def melanoma_train_dataloaders(opt):
             utils.check_dataset_balance(val_dataset_fold_balanced)
 
         return fold_loaders
+# ------------------------------------ Simple split by tfrecord ------------------------------------
     else:
-        train_files, val_files, train_classes, val_classes = train_test_split(
-            files, classes, train_size=0.8, test_size=0.2, stratify=classes, random_state=42
-        )
-        print(f"Original training set size: {len(train_files)}")
```
Comment (rottingbadger) [inline data.py:355]: The oversampling and undersampling code block has been accidentally removed and needs to be put back in before the datasets are created.

File: data.py [hunk 2]
```diff
@@ -348,24 +350,22 @@ def melanoma_train_dataloaders(opt):
...
-            train_files, train_classes = down_sampling(train_files, train_classes, downsampling_rate)
-        print(f"After downsampling: {len(train_files)}")
+        # Simple split by tfrecord: groups 0-9 for training, 10-11 for validation
+        train_df = dataset[dataset['tfrecord'].isin(range(0, 10))]
+        val_df = dataset[dataset['tfrecord'].isin([10, 11])]
+        
+        # Assuming your CSV really does have 'image_name' (no .jpg)
+        train_files = (train_df['image_name'] + '.jpg').tolist()
+        train_classes = train_df['target'].tolist()
         train_dataset = MelanomaDataset(opt, 'train', opt['dataset']['dataset_train_path'], train_files, train_classes)
```
Comment (rottingbadger) [inline data.py]: probably needs to shuffle down below the val_files val_classes code to accomodate the missing oversampling/undersampling if statements.

**Review by rottingbadger — CHANGES_REQUESTED** [review]
